pkg/provisioner/cloudprovisioner.go

```diff
@@ -183,6 +183,33 @@ func (c *CloudProvisioner) CreateVolume(
 	}
 
 	selectedAvailabilityZone := pickAvailabilityZone(accessibilityRequirements, c.GetCloud().Location)
+	accessibleTopology := []*csi.Topology{}
+	if skuName == compute.StandardSSDZRS || skuName == compute.PremiumZRS {
+		klog.V(2).Infof("diskZone(%s) is reset as empty since disk(%s) is ZRS(%s)", selectedAvailabilityZone, diskName, skuName)
+		selectedAvailabilityZone = ""
```

Comment:
Not getting used in this if block?

Reply:
Also, since we are making the selectedAvailibilityZone = "", 
are we not using the "Preferred" accessibility requirements while using ZRS, since I can see that we are only using the "Requisite" at line no. 190-192?

Reply:
A ZRS disk is accessible from all zones, so `selectedAvailabilityZone` is not applicable (this is the setting passed to the call to `CreateManagedDisk`) and is set to "".

Event though the ZRS disk is accessible from all zones, we still wanted to respect the `allowedTopology` configuration in the `StorageClass` used to provision the disk under the assumption that the customer has zone-specific storage requirements for their workload. This comes to the driver in the `ToplogyRequirements.Requisite` field and we need to reflect this in the `CreateVolumeResponse.AccessibleTopology` field. It is strictly an artificial limitation from the standpoint of the ZRS implementation.